src/app/util/ember-compatibility-functions.cpp

```diff
@@ -283,6 +283,15 @@ CHIP_ERROR ReadSingleClusterData(const SubjectDescriptor & aSubjectDescriptor, b
 
     const EmberAfCluster * attributeCluster            = nullptr;
     const EmberAfAttributeMetadata * attributeMetadata = nullptr;
+
+#if defined(CHIP_DISABLE_UNIQUE_ID) && CHIP_DISABLE_UNIQUE_ID
+    if (aPath.mEndpointId == 0 && aPath.mClusterId == chip::app::Clusters::BasicInformation::Id &&
+        aPath.mAttributeId == chip::app::Clusters::BasicInformation::Attributes::UniqueID::Id)
+    {
+        return CHIP_ERROR_IM_GLOBAL_STATUS_VALUE(UnsupportedAttributeStatus(aPath));
```

Comment:
What about wildcard expansion and other things that look at attribute metadata?  This will lead to incorrect behavior on _write_ attempts to this attribute, for example.

The right way to do this is to have an example app configured with this attribute turned off or something, not by hacking core SDK code like this.

Reply:
Ok, I see.
Do you have any suggestions on how to do this?
I was thinking about one of the solutions:
* duplicate `examples/lighting-app/lighting-common/lighting-app.zap`,
* use the template generator to remove the `UniqueId` section from the zap file if not needed (it will be more problematic to use zap_tool)

Reply:
Do you need to duplicate lighting-app?  Or could you just turn this attribute off in lighting-app, period?

Reply:
@jlatusek and @arkq 

I had a slack chat with Boris about this today to help describe why we need the app and the direction we should go in. His recommendation is that we create a new example app with the specific purpose of being a device that doesn't have a UniqueID based on configuration set in the zap file. So new example app called something like `no-unique-id-app`.

He mentioned (with some paraphrasing):
>The new app should:
> 1. Have the ClusterRevision set accordingly, to a revision where that's allowed.
> 2. Be protected from having that changed by the update_cluster_revision.py script (this can be a followup, probably).